webpages/settings/permissions.js

```diff
@@ -0,0 +1,44 @@
+const lightThemeLink = document.createElement("link");
+lightThemeLink.setAttribute("rel", "stylesheet");
+lightThemeLink.setAttribute("href", "light.css");
+
+const vue = new Vue({
+  el: "body",
+  data: {
+    screenshotPath: "../../images/screenshots/permissions-dark.png",
+  },
+  methods: {
+    msg(message, ...param) {
+      return chrome.i18n.getMessage(message, ...param);
+    },
+  },
+});
+
+chrome.storage.sync.get(["globalTheme"], function (r) {
+  let rr = false; //true = light, false = dark
+  if (r.globalTheme) rr = r.globalTheme;
+  if (rr) {
+    document.head.appendChild(lightThemeLink);
+    vue.theme = "../../images/screenshots/permissions-light.png";
+  }
+});
+
+document.title = chrome.i18n.getMessage("permissionsTitle");
+
+const promisify = (callbackFn) => (...args) => new Promise((resolve) => callbackFn(...args, resolve));
+
+document.getElementById("permissionsBtn").addEventListener("click", async () => {
+  const manifest = await (await fetch("/manifest.json")).json();
+  const origins = manifest.permissions.filter((url) => url.startsWith("https://"));
```

Comment:
Here, origins are not hardcoded. Please choose a single way.
If you want to keep this one, keep in mind the manifest is already available with `chrome.runtime.getManifest()`
It's very very unlikely we'll add more required permissions. We may however, add optional permissions like turbowarp.org